ambry-store/src/main/java/com.github.ambry.store/DiskManager.java

```diff
@@ -359,6 +359,32 @@ boolean shutdownBlobStore(PartitionId id) {
     }
   }
 
+  /**
+   * Given partition id, remove the corresponding blob store in disk manager
+   * @param id the {@link PartitionId} of the {@link BlobStore} which should be removed.
+   * @return {@code true} if store removal was successful. {@code false} if not.
+   */
+  boolean removeBlobStore(PartitionId id) {
+    BlobStore store = stores.get(id);
+    if (store == null) {
+      logger.info("Store {} is not found in disk manager", id);
+      return true;
+    }
+    if (!running || store.isStarted()) {
```

Comment:
Why is (running == false) an error condition here?

Reply:
`running` here indicates if the `DiskManager` is running. If `DiskManager` is closed, we are not supposed to remove store because current in-memory status in `DiskManager` is no longer up-to-date.